### DISCUSSION TITLE
Folder colour cycling - new colouring schemes

### DISCUSSION THREAD

**Discussion by piersmyers** [JanDeDobbeleer/oh-my-posh#3825] [Ideas]
Hi Jan,

I like the folder colour cycling feature you added recently. As it seems so easy for you to colour each folder could we have some other colouring schemes, for example:

Set current folder colour and a different colour for all its parent folders
Set current folder colour and fade parent folders (to a predefined transparency)
Cycle colours backwards (so current folder is always the same)

May be have 'Current Folder' and 'Parent Folder' properties that we can customise ourselves?

**Comment by JanDeDobbeleer** [comment]
I don't know if I would state this is easy. It could be done (probably), but I'll have to investigate what/how.

**Reply by piersmyers** [reply]
Well easy is relative, you've done most of the hard work already, just need to jiggle the colouring a bit 😜

**Reply by srushti** [reply]
One possibility (that would work for me) is to have a colour that is determined by the hash of the folder name (or the entire path). That way, most folders would have a unique colour (even if you didn't get to pick it) and you would be able to determine at a glance which one you're in. Maybe you could also configure some sort of offset to the hash globally so that you get a different set of colours, but that might hard to use.
